addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3847,6 +3851,30 @@ export class OdooEditor extends EventTarget {
                             setSelection(codeElement.firstChild, 0);
                         }
                     }
+                } else if ((shouldCreateNumberList || shouldCreateBulletList) &&
+                    !closestElement(selection.anchorNode, 'li')
+                ) {
+                    this.historyStep();
+                    const node = selection.anchorNode;
+                    const listFormatRegex = /^[\-*]\s|^\b(?:[1aA])[.)]\s*/;
+                    node.textContent = node.textContent.replace(listFormatRegex, '');
+                    fillEmpty(closestBlock(node));
+                    if (shouldCreateNumberList) {
+                        this.historyPauseSteps();
+                        this._applyCommand('toggleList', 'OL');
+                        this.historyUnpauseSteps();
+                        const deepsetNode = getDeepestPosition(selection.anchorNode, selection.anchorOffset);
+                        const closestOl = closestElement(deepsetNode[0], 'OL');
+                        if (!closestOl) return;
```

Comment:
Could we have a test case which involves reaching the deepestPosition?
Also maybe a comment too.